shell/shared/shell.js

```diff
@@ -223,10 +224,19 @@ if (Meteor.isClient) {
     filteredGrains: function () {
       var selectedApp = Session.get("selectedApp");
       var userId = Meteor.userId();
+      var result = [];
       if (selectedApp) {
         return Grains.find({userId: userId, appId: selectedApp}, {sort: {lastUsed: -1}}).fetch();
       } else {
-        return Grains.find({userId: userId}, {sort: {lastUsed: -1}}).fetch();
+        var result = Grains.find({userId: userId}, {sort: {lastUsed: -1}}).fetch()
+        var uniqueGrains = {};
+        RoleAssignments.find({}, {sort:{created:1}}).forEach(function(roleAssignment) {
+          if (!(roleAssignment.grainId in uniqueGrains)) {
+            result.push({_id : roleAssignment.grainId, title: roleAssignment.title});
```

Comment:
I think we should keep the "shared with me" files in a separate list from the user's own files. So the sidebar should have "My Files" (renamed from "All Files") followed by "Shared With Me" showing RoleAssignments. (And as discussed previously (I think), the per-app tabs should only show the user's own files.)